src/metabase/pulse.clj

```diff
@@ -63,21 +63,39 @@
                               (str "Pulse: " (:name pulse))
                               attachments)))
 
+(defn is-card-empty?
+  "Check if the card is empty"
+  [card]
+  (let [result (:result card)]
+    (or (zero? (-> result :row_count))
+        (= [[nil]]
+           (-> result :data :rows))
+        (and (= [["count"]]
+                (-> card :card :dataset_query :query :aggregation))
+             (= [[0]]
+                (-> result :data :rows))))))
+
+(defn are-all-cards-empty?
+  "Do none of the cards have any results?"
+  [results]
+  (every? is-card-empty? results))
+
 (defn send-pulse!
   "Execute and Send a `Pulse`, optionally specifying the specific `PulseChannels`.  This includes running each
    `PulseCard`, formatting the results, and sending the results to any specified destination.
 
    Example:
        (send-pulse! pulse)                       Send to all Channels
        (send-pulse! pulse :channel-ids [312])    Send only to Channel with :id = 312"
-  [{:keys [cards] :as pulse} & {:keys [channel-ids]}]
+  [{:keys [cards skip_if_empty] :as pulse} & {:keys [channel-ids]}]
```

Comment:
Clojure variables should use lisp-case, i.e. `skip-if-empty`. Of course, they'll come in from the API in snake_case (`skip_if_empty`) so there will be a few places where it will be out of your hands but everywhere else (like this `send-pulse!` function) should use lisp-case